### PR TITLE
Upload artifacts during test run

### PR DESCRIPTION
**Pull request by clee2000**
Zip and upload artifacts while run_test is running
Upgrade boto3 because I get errors about not having `botocore.vendored.six.move` if I don't

### CODE REVIEW

**Comment by clee2000** [comment]
test

**Review by huydhn — APPROVED** [review]

**Comment by huydhn** [comment]
Does this work on those non-aws runners? like ROCm or XPU?  If they need to upload artifact to GitHub first and don't have direct access to S3, then this route wouldn't work.  IMO, it's not a blocker though, just need to handle the failure accordingly

File: tools/testing/upload_artifacts.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        start = time.time()
+        try:
+            upload_to_s3_artifacts()
+            LAST_UPDATED = time.time()
+        except Exception as e:
+            print(f"Failed to upload artifacts: {e}")
+        print(f"Uploading artifacts took {time.time() - start:.2f} seconds")
+
+
+def trigger_upload_test_stats_intermediate_workflow() -> None:
```
Comment (huydhn) [inline tools/testing/upload_artifacts.py:90]: What is this func used for?  I don't see it's used anywhere, could be a copy/paste mistake

**Comment by clee2000** [comment]
> Does this work on those non-aws runners? like ROCm or XPU? If they need to upload artifact to GitHub first and don't have direct access to S3, then this route wouldn't work. IMO, it's not a blocker though, just need to handle the failure accordingly

No, it doesn't work for non aws runners.  There should be a try catch to handle when zipping and uploading fails

File: tools/testing/upload_artifacts.py (hunk 1 above)
Comment (huydhn) [reply tools/testing/upload_artifacts.py:91]: Yeah, the token would only have read access and will not be able to dispatch workflow from CI

**Pull request closed without merging**